pkg/storage/chunk/client/object_client.go

```diff
@@ -165,7 +166,7 @@ func (o *client) getChunk(ctx context.Context, decodeContext *chunk.DecodeContex
 
 	readCloser, size, err := o.store.GetObject(ctx, key)
 	if err != nil {
-		return chunk.Chunk{}, errors.WithStack(err)
+		return chunk.Chunk{}, errors.WithStack(fmt.Errorf("failed to load object '%s': %w", key, err))
```

Comment:
Can't you make this a separate error and check for it in `loadRemoteSchema`? Relying on the text of the error is not great.